### PR TITLE
Use codecov for checking coverage step instead of coveralls

### PR DESCRIPTION
**Pull request by helobinvn**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -22,19 +22,12 @@ jobs:
...
-        env:
-          CI_NAME: Github
-          CI_BUILD_NUMBER: ${{ github.run_id }}
-          CI_BUILD_URL: https://github.com/${{ github.repository }}/commit/${{ github.event.after }}/checks
-          CI_BRANCH: ${{ steps.refs.outputs.branch_name }}
-          CI_PULL_REQUEST: ${{ steps.refs.outputs.pr_number }}
-        run: |
-          mvn verify javadoc:javadoc coveralls:report -Pcoverage -DrepoToken=${{ secrets.COVERALLS_REPO_TOKEN }} -B -V
+          mvn verify javadoc:javadoc jacoco:report -Pcoverage -B -V
+      - name: Check overage in codecov
```
Comment (liry) [inline .github/workflows/build.yml:28]:
```suggestion
      - name: Upload coverage report to Codecov
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -22,19 +22,12 @@ jobs:
...
-      - name: Run build steps and check coverage
-        env:
-          CI_NAME: Github
-          CI_BUILD_NUMBER: ${{ github.run_id }}
-          CI_BUILD_URL: https://github.com/${{ github.repository }}/commit/${{ github.event.after }}/checks
-          CI_BRANCH: ${{ steps.refs.outputs.branch_name }}
-          CI_PULL_REQUEST: ${{ steps.refs.outputs.pr_number }}
-        run: |
-          mvn verify javadoc:javadoc coveralls:report -Pcoverage -DrepoToken=${{ secrets.COVERALLS_REPO_TOKEN }} -B -V
+          mvn verify javadoc:javadoc jacoco:report -Pcoverage -B -V
```
Comment (liry) [inline .github/workflows/build.yml:27]: Please also remove `coveralls-maven-plugin` plugin from `pom.xml`

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-java-merge-compliance-check/231/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/compliance/job/gooddata-java-fossa-scanning-zuul-el7/70/ : FAILURE in 1m 56s (non-voting)

**Pull request merged**